src/megaapi_impl.cpp

```diff
@@ -12905,11 +12937,38 @@ void MegaApiImpl::sendPendingRequests()
 		{
 			const char* dstFilePath = request->getFile();
             int type = request->getParamType();
-			Node *node = client->nodebyhandle(request->getNodeHandle());
+            handle h = request->getNodeHandle();
+            const char *fa = request->getText();
+            const char *base64key = request->getPrivateKey();
+
+            Node *node = client->nodebyhandle(h);
 
-			if(!dstFilePath || !node) { e = API_EARGS; break; }
+            if(!dstFilePath || (!fa && !node) || (fa && (!base64key || ISUNDEF(h))))
+            {
+                e = API_EARGS;
+                break;
+            }
 
-            e = client->getfa(node, (fatype)type);
+            string fileattrstring;
+            string key;
+            if (!fa)
+            {
+                fileattrstring = node->fileattrstring;
```

Comment:
Yes. Some MegaNode may not have been initialized with the `fileattrstring` because it's not really needed. In those cases, the request is created without the `setText()`, so `fa == NULL` and both `fileattrstring` and `nodekey` need to be copied from the node retrieved via `nodebyhandle()`.